src/entry_point.py

```diff
@@ -201,6 +256,311 @@ def _constructor_menuinst(prefix, pkg_names=None, root_prefix=None, remove=False
         install(str(json_path), remove=remove, prefix=prefix, root_prefix=root_prefix)
 
 
+def _is_subdir(directory: Path, root: Path) -> bool:
+    """
+    Helper function to detect whether a directory is a subdirectory.
+
+    Rely on Path objects rather than string comparison to be portable.
+    """
+    return directory == root or any(root == parent for parent in directory.parents)
+
+
+def _get_init_reverse_plan(
+    uninstall_prefix,
+    prefixes: list[Path],
+    for_user: bool,
+    for_system: bool,
+    anaconda_prompt: bool,
+) -> list[dict]:
+    """
+    Prepare conda init --reverse runs for the uninstallation.
+
+    Only grab the shells that were initialized by the prefix that
+    is to be uninstalled since the shells within the prefix are
+    removed later.
+    """
+    import re
+
+    from conda.base.constants import COMPATIBLE_SHELLS
+    from conda.common.compat import on_win
+    from conda.common.path import win_path_to_unix
+    from conda.core.initialize import (
+        CONDA_INITIALIZE_PS_RE_BLOCK,
+        CONDA_INITIALIZE_RE_BLOCK,
+        _read_windows_registry,
```

Comment:
Hm, this is going to be a bit brittle... I'm not sure if `conda` respects their deprecation policy for `_private` names. Do they?

Reply:
I'm not sure, but if `conda` changes this, `test_uninstallation_init_reverse` will fail and we can change the name of the function.

The alternative is to copy and paste the code. However, I want to re-use as many `conda` components as possible to be closer to `conda` behavior.

Reply:
Ok, good point, we'll catch this at build time because it's frozen anyway.